### ISSUE TITLE
Clarify ability to reason with annotations - note additional statements to add to reasoner

### ISSUE THREAD

**Issue by fhirsch** [w3c/web-annotation#98]
Issue:

Not clear how to reason with annotations as all statements are not explicit, some may need to be added to reasoner manually. 

Proposal:

Make explicit what might be added manually to reasoner.

Add a Note at the end of Section 1 in Model before 1.1 (Introduction) based on what Jacob suggested:

[[

Note: 

Implementers of reasoners can explicitly add the inference that "body relatedTo target" to enable reasoning with annotations. In addition,  inferring "motivation subclassOf relatedTo" is also a valid interpretation (or use the predicate 'annotates'), especially if it serves some search retrieval function. "

]]

**Comment by azaroth42** [comment]
-1 for several reasons.

Technically it is incorrect. A Motivation is an instance of a class, it is not a predicate. Thus inferring that a motivation is a predicate would be simply wrong. If we wanted to go that route, we would just put that exact triple in the annotation: body motivation target
Further, we tried to do this in the past by having a predicate associated with the annotation, and the feedback was that it was both strange to have a non metadata reference to a predicate and did not meet the needs of the people that wanted it, as with the cardinality of body and target, it might be impossible (no body), or wrong (multiple heterogeneous bodies or targets)

Further, the actual inference logic even if it was reasonable, would be significantly more complex than that. It would need to take into account the intended semantics of the motivation (a comment body on an annotation with motivation highlighting does not highlight the target) and the existence of hasRole on the various resources.

From a principles standpoint, we have tried hard to explicitly not talk about RDF related features and functionality in the model document to the point taking out "ontology" and other similar words. To start in the introduction to talk about reasoners and inferencing is completely counter to that stated principle.  We have #24 which even goes so far as to state the opposite of this proposal.

And editorially, introducing a somewhat normative statement in a note in the introduction before any of the concepts are described is just not the right solution, even if we were to accept that this is a real issue, against all previous discussion.

**Comment by fhirsch** [comment]
@azaroth Perhaps I misread what Jacob said.

do you have an alternative proposal to address Randall's concern, which appears legitimate?

Let's rephrase the question - are there any implicit assumptions about reasoning where a note would be helpful to the reader to make them explicit, and if so where should we put that note.

**Comment by BigBlueHat** [comment]
Good points @azaroth42.

Perhaps, then, we can include an Appendix (or some such) in the new "model only" doc that explains how to explicitly state relationships between body and target, or (alternatively) highlight that these Annotation documents aren't intended for that...and to "just use RDF."

I do think Randall's points need addressing somewhere, so that other people looking at Annotation and wondering how to (or if to) state that body and target are related are given a way to do it or an explanation of why the Annotation doesn't.

Maybe we already do that, and I've just missed it. :smiley:

**Comment by jjett** [comment]
I think it'll be difficult to address Randall's concerns about whether or
not we can interpret motivation as a relationship between the body and the
target. Part of the point I was trying to make is that any interpretation
of motivations is necessarily an idiosyncratic one because there is not a
broad consensus on what those relationships are or should be.

That said, I noticed a lot of things coming out of TPAC that were making
these kind of idiosyncratic interpretations (e.g., division of the hasBody
predicate into hasBody and hasTextBody -- this conflates the role of body
with the kind of thing that the body is (i.e., resource or literal)). This
is not very good RDF but may have value for a number of things such as
serializations and various reasoners (including those using flavors of OWL
that are not OWL Full).

I'm -0 on the proposal. I can see the value of body relatedTo target (I can
even see body annotates target) but, body motivation target is a bridge too
far.

Regards,

Jacob

P.S. I'm about to start quals at the end of the week so I'm afraid that I
won't be much help for the next few weeks.

---

Jacob Jett
Research Assistant
Center for Informatics Research in Science and Scholarship
The Graduate School of Library and Information Science
University of Illinois at Urbana-Champaign
501 E. Daniel Street, MC-493, Champaign, IL 61820-6211 USA
(217) 244-2164
jjett2@illinois.edu

**Comment by azaroth42** [comment]
I think the main point of @tilgovi's argument is:

> Interpretation of motivations that I would like to specify so that the interpretation is not idiosyncratic but normative.

Such that a triple could be inferred of Body _something_ Target, for all combinations of all 0-m bodies and 1-n targets.

I disagree that this is a legitimate concern because it implies that there is some way in which we can reliably know what to put in _something_. The motivation is intentionally fuzzy and high level because it's impossible to clearly know the interpretation (indeed, the motivation) of the user for their annotation.

If this is not idiosyncratic, then triples could simply be added to the graph that express the relationship between resources as desired.  For example, to be explicit that the body `foaf:depicts` the target: 

```
{
  "type": "Annotation",
  "body": {
    "source": "http://example.org/photo1",
    "role": "tagging",
    "foaf:depicts": "http://example.com/person1"
  },
  "target": "http://example.com/person1"
}
```

**Comment by BigBlueHat** [comment]
@tilgovi if we add a note or appendix or some such that explains the "just RDF" approach (as expressed in @azaroth42's `foaf:depicts` example), do you feel that sufficiently addresses your concern?

I agree with @azaroth42 and @jjett that turning `motivations` into something they're not (or trying to re-specify them into something "more") would really be wheel reinvention (we already have RDF that does this) and actually limit the value of the intentional vagueness of motivation (as currently specified)--which (to me at least) is more about "signaling" usage vs. defining a thing categorically (or in RDF terms "classifying").

Given that we can simply add ad hoc triples into the graph via JSON-LD constructs, I feel the use cases are all best served with our current approach:
- use `motivations` to signal what this body MAY be used for along side the target
- use RDF to further specify relationship between any part of the graph--typically between `body` and `target` and/or the annotation and either of them

Thoughts?

**Comment by tilgovi** [comment]
Rob writes:

> The motivation is intentionally fuzzy and high level because it's impossible to clearly know the interpretation (indeed, the motivation) of the user for their annotation.

What is the connection between part of statement and precision? I would think it would be equally reasonable to say about relations that they are "are intentionally fuzzy and high level" and don't understand why we come to consensus about motivations but not relations.

Just as we say nothing in the spec about what shape a body or target of an annotation motivatedBy describing has we can similarly define "describes" with range and domain as simply rdf:Resource. Such a construction would seem to me to be equally "fuzzy", which is to say as fuzzy as our definitions of "describing" or "describes".

I would not suggest that we adopt any such relations from external vocabularies for exactly this same need for flexibility through imprecision, just as we have not adopted terms from external vocabularies for the motivations.

**Comment by azaroth42** [comment]
This approach was already discussed in the roles conversation and rejected on the grounds of added confusion (not wanting to have parallel motivations and relationships) and difficulty for processing in terms of discovering the body (or target) of an annotation. It would be redundant to have:

```
{
 "type": "Annotation",
 "body": {
    "source": "http://example.org/body1",
    "role": "describing",
    "describes": "http://example.com/target1"
  },
  "target" : "http://example.com/target1"
}
```

**Comment by tilgovi** [comment]
> not wanting to have parallel motivations and relationships

I'm not suggesting to have them in parallel. I'm suggesting to have relationships instead.

> It would be redundant to have [role: describing and describes: target]

Yes. It would be. That's why I'm not suggesting role either.

I'm suggesting just to have the relationship and questioning the value of motivation (and role). I am becoming only more convinced that motivation is intentionally vague and needlessly indirect. Role suffers from the exact same issues but is more granular, letting us get specific about which vague, indirect relationship we're avoiding.

**Comment by azaroth42** [comment]
Then I propose close wontfix as we've already discussed and rejected this approach. Sorry!

**Comment by jjett** [comment]
You still run into the problem that "A describes B" is only sometimes an
annotation and is other times a description. IMO, the main reason we cannot
do this is because it is not the case the the set of all descriptions
matches or fits within the set of all annotations.

At best you could only ever say "A annotates B" OR "A describes B" but
never both. Any inference that these have the same semantics is totally
idiosyncratic and dependent upon the specific annotator.

The way the model is structured now is to say "A annotates B" AND
"Annotator mayIntendThat 'A describes B'" (assuming the optional
motivatedBy has actually been asserted, otherwise it only claims the first
thing, "A annotates B").

I get that there is a search use case that requires this weird kind of
searching by intended role of body content but the fact is, at the end of
the day, there isn't much evidence to support that we are interpreting
those intended roles correctly. Any system that tries to mandate this kind
of typing on the user-side is going to be obnoxious to them (because 1)
data entry and 2) "doesn't have my choice" (e.g. remarking not
commenting)). This kind of search use case isn't very supportable because
developers don't actually have warrant from the users to assume their
intentions (there are simply too many different, disparate end user
communities supported by too many different, disparate developer
communities). Motivation is really the best we can do and still have
annotations.

Regards,

Jacob

---

Jacob Jett
Research Assistant
Center for Informatics Research in Science and Scholarship
The Graduate School of Library and Information Science
University of Illinois at Urbana-Champaign
501 E. Daniel Street, MC-493, Champaign, IL 61820-6211 USA
(217) 244-2164
jjett2@illinois.edu

**Comment by jjett** [comment]
+1

---

Jacob Jett
Research Assistant
Center for Informatics Research in Science and Scholarship
The Graduate School of Library and Information Science
University of Illinois at Urbana-Champaign
501 E. Daniel Street, MC-493, Champaign, IL 61820-6211 USA
(217) 244-2164
jjett2@illinois.edu

**Comment by BigBlueHat** [comment]
@tilgovi that route got rejected (iirc) due to it _having_ to be overly specific and exclusive.

Here are the examples that match (more or less) what you're describing:
- https://www.w3.org/annotation/wiki/Expressing_Role_in_Multi-Body_Annotations#Role_as_Subproperty_of_hasBody.2FhasTarget
- https://www.w3.org/annotation/wiki/Expressing_Role_in_Multi-Body_Annotations#Role_as_Subproperty
- https://www.w3.org/annotation/wiki/Expressing_Role_in_Multi-Body_Annotations#Role_as_Subproperty_2

Doing things like "looping through the bodies" for output or even keeping up with a limitless number of possible relationships between body and target which MAY be considered "annotations" (but are likely more accurately RDF-style resource _description_) becomes unwieldy quickly...

**Comment by tilgovi** [comment]
Think we can close for now.

**Comment by azaroth42** [comment]
Let's confirm with WG on 11/11 call before closing :)
